compiler/rustc_attr/src/builtin.rs

```diff
@@ -217,85 +217,81 @@ where
                     let mut issue_num = None;
                     let mut is_soft = false;
                     for meta in metas {
-                        if let Some(mi) = meta.meta_item() {
```

Comment:
This code is being refactored in #93017. Does c271373db4a58920c2a9629cebaab321b3e2ca57 need to use more let-else?

Reply:
It still has a diverging block in the else case, so your refactored code can still benefit from `let else` adoption. But I can remove it here so that it doesn't cause conflicts. Does that sound like a plan to you?